### DISCUSSION TITLE
Clarification for 'Retrieve a row' vs Action 'Fetch row'

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#10450] [Help] [answered]
Looking at the following documentation https://docs.budibase.com/reference/get_tables-tableid-rows-rowid it mentions that this gets a single row enriched with the full relational rows. I noticed when using actions 'fetch row' it appears this is not the case and does in fact only get the row data without any of the relationship row data, not even the 'primaryDisplay'.

Is this the intended behavior or am I doing something incorrectly? Can I use the 'retrieve a row' functionality inside an action and then use object modifiers to focus on specific relational data? I'm looking at functionality where I pull the relational data, then use sorting or other functionality that would pick out the maximum related report report for example. Currently I am using a combination of queries and fetch row actions to retrieve and store certain data in app state that gets used in forms to prefill certain data. I have noticed that when storing object data in app state it actually is stored as a JSON string, and to retrieve particular values I then have to use JSON.parse(). It would seem that certain actions could be significantly reduced or eliminated with some more streamlining. Also, it appears that under action documentation here https://docs.budibase.com/docs/actions the 'fetch row' action is currently missing.

**Comment by melohagan** [comment, marked as answer]
Hey @mmalessa80 

I believe this is intended. The purpose of the **Fetch row** action is to retrieve a single row when you have an encoded ID and want to retrieve a row (such as an *On Change* event on a relationship picker. It's a fairly niche scenario).

If you want to retrieve related rows within a Budibase DB table there's a number of options available:

 - https://docs.budibase.com/docs/formula#fields-from-relationships
 - https://docs.budibase.com/docs/filter-by-relationships 
 - https://docs.budibase.com/docs/fetching-data#displaying-relational-data
